features/integration/simulate.feature

```diff
@@ -0,0 +1,94 @@
+Feature: Simulating transactions
+  Background:
+    Given an algod v2 client
+    And a kmd client
+    And wallet information
+    And suggested transaction parameters from the algod v2 client
+    And I create a new transient account and fund it with 10000000 microalgos.
+    And I make a transaction signer for the transient account.
+
+  @simulate
+  Scenario Outline: Simulating successful payment transactions
+    Given default transaction with parameters <amt> "<note>"
+    When I get the private key
+    And I sign the transaction with the private key
+    And I simulate the transaction
```

Comment:
Do you think it's worth distinguishing between simulating with signed transactions and simulating without ?

If so, this step could instead by:

```cucumber
And I simulate-signed the transaction
```
leaving room for the future

```cucumber
And I simulate-unsigned the transaction
```

But if the API is going to be the same (i.e. simulate just takes the transactions as they are and doesn't need special methods to handle unsigned), then we don't need to change this step.

We could also punt this to #267 and if the API changes at a later date, we would change the steps as well - though that would be a bit annoying.

Reply:
I don't think the API would change, so need for two separate steps. 

Currently, the simulate endpoint itself sends the msgpack encoded transaction (bytearray-like object). As a part of #267 , we probably need helper methods to encode unsigned transactions into msgpack, and then send that to the simulate endpoint. Currently, we have explicit checks in the SDKs to look for signatures before doing this encoding.